torch/fx/experimental/symbolic_shapes.py

```diff
@@ -329,28 +329,68 @@ def _get_key(self):
         """
         return (len(self.replacements), len(self.divisible))
 
-    # NB: This is only called for input symbolic sizes; intermediate symbolic
-    # sizes are allocated via a different mechanism
-    def create_symint(self, name, val):
-        assert val >= 0
+    def create_symbolic_sizes_strides(self, ex: torch.Tensor):
+        """
+        Returns a list of symbolic sizes and strides for the given tensor.
+        We try our best to express stride in terms of the sizes, so as to not
+        introduce new symbolic variables.
```

Comment:
Hmm, when would we be in a situation where we actually need to introduce new symbolic variables for any of the strides?

It seems like strides can/should always be expressed in terms of size variables (either from the same tensor, or from a different tensor), but maybe I'm just not thinking of an edge case where this can happen.

Reply:
You can have arbitrarily weird strides.

```
>>> a = torch.arange(1, 20)
>>> unfold_a = a.unfold(0, 5, 2)
>>> unfold_a.stride()
(2, 1)
>>> unfold_a.shape
torch.Size([8, 5])
```

Reply:
<img width="318" alt="image" src="https://user-images.githubusercontent.com/6355099/195685459-fc4ae27b-73b1-48d7-bf02-10e2ecfd76c8.png">

Reply:
ah, yep thanks - sometimes I forget how horrible striding logic can be :)